test/Sentry.Tests/Internals/MainSentryEventProcessorTests.cs

```diff
@@ -164,35 +164,45 @@ public void Process_NoReleaseOnOptions_SameAsCachedVersion()
             Assert.Equal(sut.Release, evt.Release);
         }
 
-        [Fact]
-        public void Process_EnvironmentOnOptions_SetToEvent()
+        [Theory]
+        [InlineData(null, Constants.DefaultEnvironmentSetting)] // Missing: will get default value.
+        [InlineData("", Constants.DefaultEnvironmentSetting)] // Missing: will get default value.
+        [InlineData(" ", Constants.DefaultEnvironmentSetting)] // Missing: will get default value.
+        [InlineData("a", "a")] // Provided: nothing will change.
+        [InlineData("production", "production")] // Provided: nothing will change. (value has a lower case 'p', different to default value)
+        [InlineData("aBcDe F !@#$ gHi", "aBcDe F !@#$ gHi")] // Provided: nothing will change. (Case check)
+        public void Process_EnvironmentOnOptions_SetToEvent(string environment, string expectedEnvironment)
         {
-            const string expected = "Production";
-            _fixture.SentryOptions.Environment = expected;
+            _fixture.SentryOptions.Environment = environment;
             var sut = _fixture.GetSut();
             var evt = new SentryEvent();
 
             _ = sut.Process(evt);
 
-            Assert.Equal(expected, evt.Environment);
+            Assert.Equal(expectedEnvironment, evt.Environment);
         }
 
-        [Fact]
-        public void Process_NoEnvironmentOnOptions_SameAsEnvironmentVariable()
+        [Theory]
+        [InlineData(null, Constants.DefaultEnvironmentSetting)] // Missing: will get default value.
+        [InlineData("", Constants.DefaultEnvironmentSetting)] // Missing: will get default value.
+        [InlineData(" ", Constants.DefaultEnvironmentSetting)] // Missing: will get default value.
+        [InlineData("a", "a")] // Provided: nothing will change.
+        [InlineData("production", "production")] // Provided: nothing will change. (value has a lower case 'p', different to default value)
```

Comment:
Given the privious suggestion change:

```suggestion
        [InlineData("Production", "Production")] // Provided: nothing will change. (value has a upper case 'p', different to default value)
```

Reply:
I don't understand the suggestion here? I was trying to check that the word is "the same" but the casing different .. which means, we don't override and accept the user-defined value.

The comment I made there is wrong though. I'll need to fix it to: // Provided: nothing will change. (value has a lower case 'p', different to default value)

Reply:
there's nothing wrong here, I just suggested above we use `production` instead of `Production` so your test here would need to flip: `Production` would mean user-defined value, because `production` will be the new default

Reply:
Yep - totally. I misread. yep. 💯